Add validation options for item_hints

DobbyDigital opened the pull request:
- Adds the list of all item names from ItemList as choices to item_hints
- This prevents the plando validation tool from crashing if any items are supplied in that list in the plando

DobbyDigital committed:
Add validation options for item_hints

- Adds the list of all item names from StartingItems as choices to item_hints
- This prevents the plando validation tool from crashing if any items are supplied in that list in the plando

DobbyDigital committed:
Quickly fix the spacing before Zannick notices, and pretend you did it right the first time

fenhl commented:
Is this the right list? When I looked at it earlier, it seemed to take item names from `ItemList`.

DobbyDigital commented:
ItemList has some extra stuff that is probably not valid (like the events), whereas I think that the everything table just had valid stuff. Though I will double-check, especially for weird things like trade items

DobbyDigital commented:
Hmm.. looks like it doesn't include that stuff. So probably, to be safe, will switch to use the stuff from the ItemList if the type is 'Item'. I don't know why you'd want an item hint for 'Bombs 5', but you do you fam

DobbyDigital committed:
Change to use ItemList instead

- Use ItemList instead of everything for valid options.
- There are some options which don't have the item tag which may be valid (such as "Boss Key (Forest Temple)"), but in general the only tag for which all items work is the Item tag

Zannick commented:
> Quickly fix the spacing before Zannick notices, and pretend you did it right the first time

you missed one <3
> ```py
>         name           = 'item_hints',
>         type           =  list,
> ```
